### PR TITLE
pcds-5.1.1

### PR DESCRIPTION
**Pull request by ZLLentz**
Automated release notes between 5.1.0 and 5.1.1 (note: 5.1.0 was dropped due to a yanked package)
---------------------------------------
Added the Following Packages
----------------------------

- pip-audit
- pipdeptree
- suitcase-csv
- suitcase-json-metadata
- suitcase-jsonl
- suitcase-specfile
- suitcase-tiff
- tiled

Added the Following Dependencies
--------------------------------

- anyio (required by httpcore, starlette)
- asgiref (required by uvicorn)
- cachecontrol (required by pip-audit)
- cachey (required by tiled)
- cyclonedx-python-lib (required by pip-audit)
- ecdsa (required by python-jose)
- fastapi (required by tiled)
- h11 (required by httpcore, uvicorn)
- html5lib (required by beautifulsoup4, lxml, pip-audit, sphinxcontrib-htmlhelp)
- httpcore (required by httpx)
- httpx (required by tiled)
- jmespath (required by tiled)
- lockfile (required by cachecontrol, pip-audit)
- lz4 (required by fonttools, tiled)
- orjson (required by tiled)
- packageurl-python (required by cyclonedx-python-lib)
- pip-api (required by pip-audit)
- progress (required by pip-audit)
- pyarrow (required by fsspec, intake, papermill, tiled)
- pydantic (required by fastapi, tiled)
- python-jose (required by tiled)
- python-multipart (required by tiled)
- requirements-parser (required by cyclonedx-python-lib)
- resolvelib (required by pip-audit)
- rfc3986 (required by httpx)
- rsa (required by python-jose)
- starlette (required by fastapi)
- typer (required by tiled)
- types-setuptools (required by cyclonedx-python-lib)
- types-toml (required by cyclonedx-python-lib)
- uvicorn (required by tiled)
- watchgod (required by tiled)

PCDS Package Updates
--------------------

|   Package   |  Old  |  New  |                       Release_Notes                        |
|:-----------:|:-----:|:-----:|:----------------------------------------------------------:|
| pcdsdevices | 5.0.0 | 5.0.2 | https://github.com/pcdshub/pcdsdevices/releases/tag/v5.0.2 |
|  tc_release | 0.2.1 | 0.2.2 | https://github.com/pcdshub/tc_release/releases/tag/v0.2.2  |
|    typhos   | 2.1.0 | 2.2.0 |   https://github.com/pcdshub/typhos/releases/tag/v2.2.0    |

Lab Community Package Updates
-----------------------------

|  Package   |  Old  |   New    |
|:----------:|:-----:|:--------:|
| databroker | 1.2.4 | 2.0.0a18 |

Python Community Core Package Updates
-------------------------------------

| Package |  Old   |  New   |
|:-------:|:------:|:------:|
| ipython | 7.29.0 | 7.30.1 |

Other Python Community Major Updates
------------------------------------

|   Package    |  Old   |  New   |
|:------------:|:------:|:------:|
|   colorcet   | 2.0.6  | 3.0.0  |
| cryptography | 35.0.0 | 36.0.0 |
|   pymongo    | 3.12.1 |  4.0   |
|  setuptools  | 59.1.1 | 50.3.2 |

Automated release notes between 5.0.0 and 5.1.1 (note: 5.1.0 was dropped due to a yanked package)
--------------------------------

Added the Following Packages
----------------------------

- atlassian-python-api
- pip-audit
- pipdeptree
- qrcode
- suitcase-csv
- suitcase-json-metadata
- suitcase-jsonl
- suitcase-specfile
- suitcase-tiff
- tiled

Added the Following Dependencies
--------------------------------

- aiosignal (required by aiohttp)
- anyio (required by httpcore, starlette)
- aom (required by ffmpeg)
- asgiref (required by uvicorn)
- c-blosc2 (required by imagecodecs)
- cachecontrol (required by pip-audit)
- cachey (required by tiled)
- cyclonedx-python-lib (required by pip-audit)
- ecdsa (required by python-jose)
- fastapi (required by tiled)
- fonttools (required by matplotlib, matplotlib-base)
- freeglut (required by jasper)
- frozenlist (required by aiohttp, aiosignal)
- h11 (required by httpcore, uvicorn)
- html5lib (required by beautifulsoup4, lxml, pip-audit, sphinxcontrib-htmlhelp)
- httpcore (required by httpx)
- httpx (required by tiled)
- jmespath (required by tiled)
- libdrm (required by libva)
- libglu (required by jasper)
- libpciaccess (required by libdrm)
- libva (required by ffmpeg)
- libvpx (required by ffmpeg)
- lockfile (required by cachecontrol, pip-audit)
- lz4 (required by fonttools, tiled)
- mesalib (required by libva)
- munkres (required by fonttools)
- orjson (required by tiled)
- packageurl-python (required by cyclonedx-python-lib)
- pip-api (required by pip-audit)
- progress (required by pip-audit)
- pyarrow (required by fsspec, intake, papermill, tiled)
- pydantic (required by fastapi, tiled)
- python-jose (required by tiled)
- python-multipart (required by tiled)
- python-tzdata (required by pytz-deprecation-shim)
- pytz-deprecation-shim (required by tzlocal)
- requirements-parser (required by cyclonedx-python-lib)
- resolvelib (required by pip-audit)
- rfc3986 (required by httpx)
- rsa (required by python-jose)
- setuptools-scm (required by lmfit)
- starlette (required by fastapi)
- svt-av1 (required by ffmpeg)
- typer (required by tiled)
- types-setuptools (required by cyclonedx-python-lib)
- types-toml (required by cyclonedx-python-lib)
- uvicorn (required by tiled)
- watchgod (required by tiled)
- x265 (required by ffmpeg)
- xorg-fixesproto (required by xorg-libxfixes)
- xorg-inputproto (required by xorg-libxi)
- xorg-libxfixes (required by freeglut, xorg-libxi)
- xorg-libxi (required by freeglut)

Removed the Following Packages
------------------------------

- inflection
- sleef

PCDS Package Updates
--------------------

|   Package    |  Old   |  New   |                        Release_Notes                         |
|:------------:|:------:|:------:|:------------------------------------------------------------:|
|    happi     | 1.10.0 | 1.10.1 |    https://github.com/pcdshub/happi/releases/tag/v1.10.1     |
| hutch-python | 1.12.0 | 1.13.0 | https://github.com/pcdshub/hutch-python/releases/tag/v1.13.0 |
| pcdsdevices  | 4.8.0  | 5.0.2  |  https://github.com/pcdshub/pcdsdevices/releases/tag/v5.0.2  |
|  pcdsutils   | 0.5.0  | 0.6.0  |   https://github.com/pcdshub/pcdsutils/releases/tag/v0.6.0   |
| pcdswidgets  | 0.5.1  | 0.6.0  |  https://github.com/pcdshub/pcdswidgets/releases/tag/v0.6.0  |
|    pytmc     | 2.10.0 | 2.11.0 |    https://github.com/pcdshub/pytmc/releases/tag/v2.11.0     |
|  tc_release  | 0.2.1  | 0.2.2  |  https://github.com/pcdshub/tc_release/releases/tag/v0.2.2   |
|    typhos    | 2.0.0  | 2.2.0  |    https://github.com/pcdshub/typhos/releases/tag/v2.2.0     |
|  whatrecord  | 0.1.3  | 0.2.1  |  https://github.com/pcdshub/whatrecord/releases/tag/v0.2.1   |

Lab Community Package Updates
-----------------------------

|   Package    |  Old  |   New    |
|:------------:|:-----:|:--------:|
|   bluesky    | 1.8.0 |  1.8.1   |
| bluesky-live | 0.0.7 |  0.0.8   |
|  databroker  | 1.2.3 | 2.0.0a18 |
|    ophyd     | 1.6.2 |  1.6.3   |

Python Community Core Package Updates
-------------------------------------

| Package |  Old   |  New   |
|:-------:|:------:|:------:|
|  flake8 | 3.9.2  | 4.0.1  |
| ipython | 7.28.0 | 7.30.1 |
|  sphinx | 4.2.0  | 4.3.1  |
|  xarray | 0.19.0 | 0.20.1 |

Other Python Community Major Updates
------------------------------------

|      Package      |   Old    |  New   |
|:-----------------:|:--------:|:------:|
|    argon2-cffi    |  20.1.0  | 21.1.0 |
|   async-timeout   |  3.0.1   | 4.0.1  |
|      cfitsio      |  3.470   | 4.0.0  |
|      colorcet     |  2.0.6   | 3.0.0  |
|      coverage     |   5.5    |  6.2   |
|    cryptography   |  3.4.7   | 36.0.0 |
|     decorator     |  4.4.2   | 5.1.0  |
|      harfbuzz     |  2.9.1   | 3.1.1  |
|   humanfriendly   |   9.2    |  10.0  |
|       jasper      | 1.900.1  | 2.0.33 |
|        lerc       |  2.2.1   |  3.0   |
|      pymongo      |  3.12.0  |  4.0   |
|     pyopenssl     |  20.0.1  | 21.0.0 |
|     pyparsing     |  2.4.7   | 3.0.6  |
|   pytest-timeout  |  1.4.2   | 2.0.1  |
|       pyyaml      |  5.4.1   |  6.0   |
|     setuptools    |  58.0.4  | 50.3.2 |
| typing-extensions | 3.10.0.2 | 4.0.1  |
| typing_extensions | 3.10.0.2 | 4.0.1  |
|       tzdata      |  2021b   | 2021e  |
|      tzlocal      |   3.0    |  4.1   |

Original text:
--------------
Hopefully the unit tests pass
I think this failed to pick up the pre-release version of databroker though

### CODE REVIEW

**Comment by ZLLentz** [comment]
The important tests pass, but I'll run it again with a prospective fix for picking up the databroker alpha build

**Comment by ZLLentz** [comment]
Current failures are:
- `blark` master branch, probably a lot of WIP
- `pyca` installation on py310

**Comment by ZLLentz** [comment]
I think everything here is good now and the release notes look correct

File: envs/pcds/conda-packages.txt [hunk 1]
```diff
@@ -28,7 +27,7 @@ happi>=1.10.1
 holoviews>=1.10.9
 hutch-python>=1.13.0
 hxrsnd>=0.2.6
-ipython>=7.1
+ipython>=7.30.1
```
Comment (klauer) [inline envs/pcds/conda-packages.txt:30]: ✅

**Review by klauer — APPROVED** [review]
Packages look good. Some random observations:
* A whole slough of web-related things - starlette, httpx, uvicorn, h11)
* apischema competitor-of-sorts pydantic
* Some entirely new to me: watchgod, python-jose 😁 

Random thought: Would be neat to see where dependencies come in when we add new packages.

And finally, apologies for blark master failures - I'll get around to that one soon.

File: scripts/release_notes_table.py [hunk 2]
```diff
@@ -241,12 +241,12 @@ def audit_package_lists(path):
         )
 
 def main(args):
```
[the remark this answers is not in the document]
Comment (ZLLentz) [reply scripts/release_notes_table.py:243]: I'll hit this in the next go. It's too tiring to wait for the suite to churn.

**Comment by ZLLentz** [comment]
> Would be neat to see where dependencies come in when we add new packages

It should be possible to create a diagram to upload using one of the dependency mapping projects. I guess we'd seed it with all the elements in "Added the following packages"? Alternatively, it should be possible to use one of these tools to list out (required by x) for each line there.

**Comment by klauer** [comment]
>  I guess we'd seed it with all the elements in "Added the following packages"? Alternatively, it should be possible to use one of these tools to list out (required by x) for each line there.

I think either could work. It ~would~ may be be preferable to only show `new package -> new dependency (-> new sub-dependency)` and not include any already-known dependencies.

**Comment by ZLLentz** [comment]
I think the simplest to implement will be (fake info here):

Added the Following Packages
----------------
- aiosignal (required by tiled)
- anyio (required by aiosignal)
- tiled (no deps/found in package lists, write a short custom desc)

Perhaps separating the dependency packages from the intentional packages.
Note that this gets confusing when we start to mix pypi and conda...

**Comment by ZLLentz** [comment]
Dependencies added, but apparently the method I used does not work for non-python conda packages. I think a little bit more is needed to handle those...

**Comment by ZLLentz** [comment]
The overall issue here is that:
- conda knows only about conda packages
- pkg_resources knows only about python packages

This means that:
- packages installed by pypi cannot have their dependencies inspected via conda
- non-python packages installed via conda cannot have their dependencies inspected via pkg_resources

I have an approach that augments the existing code with `mamba repoquery` but it's taking a while to run- I'll probably accept this approach so long as the output looks good.

**Comment by ZLLentz** [comment]
Ok I realized a (in retrospect, simple) way to make this faster for the specific case of pumping out the release notes. But I'm going to leave some remnants of the previous slow approach with the intention of many dependency graphs in the future.

**Comment by klauer** [comment]
That makes sense, @ZLLentz. I think what you have here is really nice, even with that caveat. Good work!

**Comment by ZLLentz** [comment]
I think I fixed up the remaining table issues, except for identification of "new" packages that are dependencies of other "new" packages. I think I will hand edit these for now.

The build needs to be re-run with `pipdeptree` and `pip-audit`, the release notes need to be generated one more time, and then the env can be deployed on Monday.

**Pull request merged**